src/client/components/equalizer/equalizer.js

```diff
@@ -76,140 +55,138 @@ class Equalizer extends Component {
       navigator.mediaDevices.getUserMedia({
         audio: true,
       })
-        .then(stream => this.createSoundStream(stream))
-        .catch((e) => {
-          throw new Error(e);
+        .then(
+          this.createSoundStream(),
+        )
+        .catch((errorStream) => {
+          throw new Error(errorStream);
         });
     } else {
       throw new Error('browser doesnt support audio API');
     }
   }
 
-  widthMerge = (eventFromInputTypeRange) => {
-    const { mergeCanvasWidth: mergeCanvasWidthAsProp } = this.props;
-    mergeCanvasWidthAsProp(eventFromInputTypeRange);
-  }
-
-  playSoundFromFile = () => {
+  uploadSoundInfoFromFile = (e) => {
+    const [file] = e.target.files;
+    const audioFile = new Audio(URL.createObjectURL(file));
+    const sound = new Pizzicato.Sound({
+      source: 'file',
+      options: {
+        path: audioFile.src,
+        loop: true,
+      },
+    }, () => this.createSoundInfoInState(sound, file));
+  };
+
+  createSoundInfoInState = (sound, file) => {
     const { audioData } = this.props;
```

Comment:
You could also destruct analyzer here, like that const {audioData: {analyzer }}= this.props

Reply:
nice)